### PR TITLE
Hatch-462: Limit amount of text displayed in grid

### PR DESCRIPTION
**Pull request by luca-bitovi**
# Description

`string` and `text` fields now accept a `maxRenderLength` option for truncating strings in the UI.

# Jira ticket links

[HATCH-462](https://bitovi.atlassian.net/browse/HATCH-462)

# Test Plan

Add a `maxRenderLength` to any `string` or `text` field in a schema and observe the truncation that occurs on the frontend when a value for that field that exceeds the `maxRenderLength` is rendered.

# Definition of done

- [x] Does new code have 90% testing coverage (100% for `core`) of both unit and E2E tests?
- [x] Is code secure? If applicable, add security notes to the description.
- [x] Do all new TODO comments have Jira links with enough information?
- [ ] Were all changes published and deployed to [the grid](https://github.com/bitovi/hatchify-grid-demo)?
- [x] Has the browser error-console been reviewed to not contain new errors introduced by these code changes?
- [x] The site looks “good” above 1000px width.
- [x] The site looks “good” when the window height is small. No double scrollbars.
- [x] Were the changes tested to ensure 508 compliance?

[HATCH-462]: https://bitovi.atlassian.net/browse/HATCH-462?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/dataTypes/string/finalizeOrm.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export function finalizeOrm({ sequelize }: PartialStringORM): FinalStringORM {
       primaryKey: !!sequelize.primaryKey,
       defaultValue: sequelize.defaultValue ?? null,
       unique: !!sequelize.unique || !!sequelize.primaryKey,
+      maxDisplayLength: sequelize.maxDisplayLength ?? null,
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/string/finalizeOrm.ts]: Done

File: packages/core/src/dataTypes/string/getPartialOrm.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export function getPartialOrm(
       primaryKey: props?.primary,
       defaultValue: props?.default,
       unique: props?.unique,
+      maxDisplayLength: props?.maxDisplayLength,
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/string/getPartialOrm.ts]: Done

File: packages/core/src/dataTypes/string/types.ts [hunk 3]
```diff
@@ -17,12 +18,17 @@ export interface PartialStringControlType<TRequired extends boolean>
   min?: number
   max?: number
   regex?: RegExp
+  maxDisplayLength?: number | null
 }
 
 export interface PartialStringORM {
-  sequelize: PartialSequelizeDataType<number[] | string[], string>
+  sequelize: PartialSequelizeDataType<number[] | string[], string> & {
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/string/types.ts]: Done

File: packages/core/src/dataTypes/string/types.ts [hunk 4]
```diff
@@ -17,12 +18,17 @@ export interface PartialStringControlType<TRequired extends boolean>
...
+    maxDisplayLength?: number
+  }
 }
 
 export interface FinalStringORM {
-  sequelize: Required<PartialSequelizeDataType<number[] | string[], string>>
+  sequelize: Required<PartialSequelizeDataType<number[] | string[], string>> & {
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/string/types.ts]: Done

File: packages/core/src/dataTypes/text/finalizeOrm.ts [hunk 5]
```diff
@@ -8,6 +8,7 @@ export function finalizeOrm({ sequelize }: PartialTextORM): FinalTextORM {
       primaryKey: !!sequelize.primaryKey,
       defaultValue: sequelize.defaultValue ?? null,
       unique: !!sequelize.unique || !!sequelize.primaryKey,
+      maxDisplayLength: sequelize.maxDisplayLength ?? null,
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/text/finalizeOrm.ts]: Done

File: packages/core/src/dataTypes/text/getPartialOrm.ts [hunk 6]
```diff
@@ -10,6 +10,7 @@ export function getPartialOrm(
       primaryKey: props?.primary,
       defaultValue: props?.default,
       unique: props?.unique,
+      maxDisplayLength: props?.maxDisplayLength,
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/text/getPartialOrm.ts]: Done

File: packages/core/src/dataTypes/text/types.ts [hunk 7]
```diff
@@ -16,9 +16,12 @@ export interface PartialTextORM {
     primaryKey?: boolean
     defaultValue?: string | (() => string) | null
     unique?: boolean
+    maxDisplayLength?: number | null
   }
 }
 
 export interface FinalTextORM {
-  sequelize: Required<PartialTextORM["sequelize"]>
+  sequelize: Required<PartialTextORM["sequelize"]> & {
```
[the remark this answers is not in the document]
Comment (luca-bitovi) [reply packages/core/src/dataTypes/text/types.ts]: Done

**Review by CMLabay — APPROVED** [review]

**Comment by justinbmeyer** [comment]
Couldn't this be solved with CSS?

**Comment by luca-bitovi** [comment]
>Couldn't this be solved with CSS?

@justinbmeyer see [this Slack message](https://bitovi.slack.com/archives/C03K651NHRB/p1702417547829619?thread_ts=1701295545.249189&cid=C03K651NHRB) and the surrounding thread for our reasoning as to why CSS wasn't the best solution in this case.

**Pull request merged**